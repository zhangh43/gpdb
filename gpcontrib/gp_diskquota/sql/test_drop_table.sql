-- Test Drop table
CREATE SCHEMA sdrtbl;
SELECT diskquota.set_schema_quota('sdrtbl', '1 MB');
SET search_path TO sdrtbl;
CREATE TABLE a(i INT);
CREATE TABLE a2(i INT);
INSERT INTO a SELECT generate_series(1,100);
-- expect insert fail
INSERT INTO a SELECT generate_series(1,100000000);
-- expect insert fail
INSERT INTO a2 SELECT generate_series(1,100);
DROP TABLE a;
SELECT pg_sleep(5);
INSERT INTO a2 SELECT generate_series(1,100);

DROP TABLE a2;
RESET search_path;
DROP SCHEMA sdrtbl;


