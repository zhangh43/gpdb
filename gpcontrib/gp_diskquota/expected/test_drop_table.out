-- Test Drop table
CREATE SCHEMA sdrtbl;
SELECT diskquota.set_schema_quota('sdrtbl', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SET search_path TO sdrtbl;
CREATE TABLE a(i INT);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE a2(i INT);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO a SELECT generate_series(1,100);
-- expect insert fail
INSERT INTO a SELECT generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:sdrtbl
-- expect insert fail
INSERT INTO a2 SELECT generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name:sdrtbl
DROP TABLE a;
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

INSERT INTO a2 SELECT generate_series(1,100);
DROP TABLE a2;
RESET search_path;
DROP SCHEMA sdrtbl;
